pa/biGraph.ml

```diff
@@ -0,0 +1,176 @@
+open Pervasives
+open BatPervasives
+open Apak
+
+include Log.Make(struct let name = "BiGraph" end)
+
+module type Vertex = sig
+  type t
+  val hash : t -> int
+  val equal : t -> t -> bool
+  val compare : t -> t -> int
+  val pp : Format.formatter -> t -> unit
+end
+
+module type S = sig
+  type t
+  type vertex
+
+  val pp : Format.formatter -> t -> unit
+
+  val empty : t
+  val make : vertex BatEnum.t -> vertex BatEnum.t -> (vertex * vertex) BatEnum.t -> t
+  val max_matching : t -> int
+  val u_size : t -> int
+  val v_size : t -> int
+  val e_size : t -> int
+end
+
+module Make (V : Vertex) = struct
+  type vertex = V.t
+  type edge = V.t * V.t [@@deriving ord]
+
+  module VertexSet = BatSet.Make(V)
+  module EdgeSet = BatSet.Make(struct
+      type t = edge [@@deriving ord]
+    end)
+
+  type t =
```

Comment:
This representation isn't really used: in max_matching you immediately compute an adjacency map representation.  No need to store this representation.